### ISSUE TITLE
[JENKINS-46051] Gradle Plugin cause dead loop with using ant plugin

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/gradle-plugin#783]
<p>gradle plugins would cause dead loop with using ant and gradle build when viewing the console output.</p>

<p>the gradle plugins when loading the outline , its use </p>

<pre>
document.getElementById(<span class="code-quote">"console-outline-body"</span>)</pre>
<p>and it has the same id with ant。 </p>

<pre>
outline = document.getElementById(<span class="code-quote">"console-outline-body"</span>)
.getElementsByTagName(<span class="code-quote">'ul'</span>)[0];</pre>
<p>when I use ant first ，ant genreate the outline, but it doesnt has a ul tag.</p>

<p>so the outline is always undefined，and cause a dead loop;</p>

<p> </p>

<p><tt>the source code of the grade plugins </tt></p>

<p><tt><font color="#333333"><sup><sub><a href="https://github.com/jenkinsci/gradle-plugin" class="external-link" target="_blank" rel="nofollow noopener">gradle-plugin</a>/[src|https://github.com/jenkinsci/gradle-plugin/tree/master/src]/[main|https://github.com/jenkinsci/gradle-plugin/tree/master/src/main]/[resources|https://github.com/jenkinsci/gradle-plugin/tree/master/src/main/resources]/[hudson|https://github.com/jenkinsci/gradle-plugin/tree/master/src/main/resources/hudson]/[plugins|https://github.com/jenkinsci/gradle-plugin/tree/master/src/main/resources/hudson/plugins]/[gradle|https://github.com/jenkinsci/gradle-plugin/tree/master/src/main/resources/hudson/plugins/gradle]/<b>GradleTaskNote</b>/script.js</sub></sup></font></tt></p>

<pre>
<span class="code-keyword">if</span> (!loading) { loading = <span class="code-keyword">true</span>; <span class="code-keyword">var</span> u = <span class="code-keyword">new</span> Ajax.Updater(document.getElementById(<span class="code-quote">"side-panel"</span>), rootURL + <span class="code-quote">"/descriptor/hudson.plugins.gradle.GradleTaskNote/outline"</span>, {insertion: Insertion.Bottom, onComplete: function() { <span class="code-keyword">if</span> (!u.success()) <span class="code-keyword">return</span>; <span class="code-comment">// we can<span class="code-quote">'t us onSuccess because that kicks in before onComplete outline = document.getElementById(<span class="code-quote">"console-outline-body"</span>) .getElementsByTagName('</span>ul')[0]; loading = <span class="code-keyword">false</span>; queue.each(handle); }}); }</span>
</pre>
<p> </p>

<p> </p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kyle_ou">kyle_ou</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-46051" target="_blank">Gradle Plugin cause dead loop with using ant plugin</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/wolfs.jpg" title="wolfs's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=wolfs">wolfs</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: gradle-plugin
<li><b>label(s)</b>: plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2017-09-26T20:09:36+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 4
<li><b>imported</b>: 20251218-222106
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>gradle plugins would cause dead loop with using ant and gradle build when viewing the console output.</p>

<p>the gradle plugins when loading the outline , its use </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">document.getElementById(<span class="code-quote">"console-outline-body"</span>)</pre>
</div></div>
<p>and it has the same id with ant。 </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">outline = document.getElementById(<span class="code-quote">"console-outline-body"</span>)
.getElementsByTagName(<span class="code-quote">'ul'</span>)[0];</pre>
</div></div>
<p>when I use ant first ，ant genreate the outline, but it doesnt has a ul tag.</p>

<p>so the outline is always undefined，and cause a dead loop;</p>

<p> </p>

<p><tt>the source code of the grade plugins </tt></p>

<p><tt><font color="#333333"><sup><sub><a href="https://github.com/jenkinsci/gradle-plugin" class="external-link" target="_blank" rel="nofollow noopener">gradle-plugin</a>/[src|https://github.com/jenkinsci/gradle-plugin/tree/master/src]/[main|https://github.com/jenkinsci/gradle-plugin/tree/master/src/main]/[resources|https://github.com/jenkinsci/gradle-plugin/tree/master/src/main/resources]/[hudson|https://github.com/jenkinsci/gradle-plugin/tree/master/src/main/resources/hudson]/[plugins|https://github.com/jenkinsci/gradle-plugin/tree/master/src/main/resources/hudson/plugins]/[gradle|https://github.com/jenkinsci/gradle-plugin/tree/master/src/main/resources/hudson/plugins/gradle]/<b>GradleTaskNote</b>/script.js</sub></sup></font></tt></p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-keyword">if</span> (!loading) { loading = <span class="code-keyword">true</span>; <span class="code-keyword">var</span> u = <span class="code-keyword">new</span> Ajax.Updater(document.getElementById(<span class="code-quote">"side-panel"</span>), rootURL + <span class="code-quote">"/descriptor/hudson.plugins.gradle.GradleTaskNote/outline"</span>, {insertion: Insertion.Bottom, onComplete: function() { <span class="code-keyword">if</span> (!u.success()) <span class="code-keyword">return</span>; <span class="code-comment">// we can<span class="code-quote">'t us onSuccess because that kicks in before onComplete outline = document.getElementById(<span class="code-quote">"console-outline-body"</span>) .getElementsByTagName('</span>ul')[0]; loading = <span class="code-keyword">false</span>; queue.each(handle); }}); }</span>
</pre>
</div></div>
<p> </p>

<p> </p></pre>
</details>
</details>
<details><summary><i>2 attachments</i></summary>

- [jenkins_ant_layout.png](https://issues.jenkins.io/secure/attachment/39891/jenkins_ant_layout.png)
  > ![jenkins_ant_layout.png](https://issues.jenkins.io/secure/attachment/39891/jenkins_ant_layout.png)
- [jenkins_error.png](https://issues.jenkins.io/secure/attachment/39856/jenkins_error.png)
  > ![jenkins_error.png](https://issues.jenkins.io/secure/attachment/39856/jenkins_error.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-46051?focusedId=315092&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315092">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Stefan Wolf
Path:
 src/main/resources/hudson/plugins/gradle/GradleTaskNote/DescriptorImpl/outline.jelly
 src/main/resources/hudson/plugins/gradle/GradleTaskNote/script.js
 src/main/resources/hudson/plugins/gradle/GradleTaskNote/style.css
<a href="http://jenkins-ci.org/commit/gradle-plugin/319f42a30a616dd2def1337776eec40a4c0d10df" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/gradle-plugin/319f42a30a616dd2def1337776eec40a4c0d10df</a>
Log:
  Fix endless loop when used with Ant builder</p>

<p><span class="error">&#91;FIXES JENKINS-46051&#93;</span></p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Stefan Wolf<br/>
Path:<br/>
 src/main/resources/hudson/plugins/gradle/GradleTaskNote/DescriptorImpl/outline.jelly<br/>
 src/main/resources/hudson/plugins/gradle/GradleTaskNote/script.js<br/>
 src/main/resources/hudson/plugins/gradle/GradleTaskNote/style.css<br/>
<a href="http://jenkins-ci.org/commit/gradle-plugin/319f42a30a616dd2def1337776eec40a4c0d10df" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/gradle-plugin/319f42a30a616dd2def1337776eec40a4c0d10df</a><br/>
Log:<br/>
  Fix endless loop when used with Ant builder</p>

<p><span class="error">&#91;FIXES <a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/46051">JENKINS-46051</a>&#93;</span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>webwart:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-46051?focusedId=315465&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315465">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=wolfs" class="user-hover" rel="wolfs">wolfs</a> It definitely looks much better now!</p>

<p>There is one thing I noticed, but maybe it was already existing before: The layout of the executed ant tasks seems a little out of place: the list bullets are located outside the box, the layout of the gradle task box is fine.</p>

<p>Thanks a lot, now the console output of a running task is usable again!</p>

<p>Greetings from Berlin!</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/39891/39891_jenkins_ant_layout.png" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=wolfs" class="user-hover" rel="wolfs">wolfs</a> It definitely looks much better now!</p>

<p>There is one thing I noticed, but maybe it was already existing before: The layout of the executed ant tasks seems a little out of place: the list bullets are located outside the box, the layout of the gradle task box is fine.</p>

<p>Thanks a lot, now the console output of a running task is usable again!</p>

<p>Greetings from Berlin!</p>

<p><span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/39891/jenkins_ant_layout.png" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/wolfs.jpg" title="wolfs's avatar" /> wolfs:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-46051?focusedId=315655&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315655">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Closing since I released 1.28 containing a fix for this issue.</p>
</pre>
</details>

</ul>
</details>

<p>Closing since I released 1.28 containing a fix for this issue.</p>
